Updates for Arch, Fedora and RHEL

ne9z opened the pull request:
Clean up unsupported boot environment manager and upgrade RHEL to major version 9.

ne9z commented:
@gmelikov

gmelikov commented:
@ne9z feel free to ping me when it's ready, looks like you've added changes since last mention.

ne9z commented:
@gmelikov That's very considerate of you, thanks!  I've squashed the commits and they are ready for merge now.

By the way, I've added Root on ZFS support for Alpine Linux in another pull request.

ne9z commented:
@gmelikov Hello, I think both this PR and #331 are ready for merge now.

gmelikov commented on docs/Getting Started/NixOS/Root on ZFS/2-system-installation.rst:
```diff
@@ -0,0 +1,132 @@
+.. highlight:: sh
```
Minor note for future - file rename will lead to different link, if it's possible - we should leave same links to minimize 404s on external links to our docs.

gmelikov reviewed (approved):
Thank you, will merge as is.

cbc02009 commented:
Just out of curiosity, why was the UUID removed from arch linux? Did something change that made it no longer necessary?

Why were the commands to create a subvolume for the user's home removed?

ne9z commented:
@cbc02009

UUID was removed as part of an effort to simplify the guide:  the guide was becoming more and more bloated and the commands ugly to look at.

The new principle is to do a bare minimum of configuration here at this repo and refer users to wiki articles where necessary.

Datasets for /home is brought back by a pending pull request: #333 @gmelikov Can you review it?
